lib/main.dart

```diff
@@ -29,181 +26,67 @@ class MyApp extends StatelessWidget {
 }
 
 class MyAppState extends ChangeNotifier {
-  // whenever got changes, notify the listeners
-  var current = WordPair.random();
-  void getNext() {
-    current = WordPair.random();
+  var currentWord = WordPair.random();
+
+  void newWord() {
+    currentWord = WordPair.random();
     notifyListeners();
   }
 
-  var favorites = <WordPair>[];
+  List<WordPair> favourite = [];
```

Comment:
previously, the like button wasnt working not because we were using other types of array, but the array size was fixed where we used
```
List<WordPair> favourite = List<WordPair>.empty();
```
`List<WordPair>.empty();` is fixed. Hence, we cannot add anything to the array, then the button icon does not change when clicked